test/datatypes/datetime.coffee

```diff
@@ -1531,13 +1544,13 @@ describe 'DateTime.sameOrBefore', ->
     DateTime.parse('2000-01-31T23:59:59.999').sameOrBefore(DateTime.parse('2000-02')).should.be.true()
     DateTime.parse('2000-12-31T23:59:59.999').sameOrBefore(DateTime.parse('2001')).should.be.true()
 
-  it 'should accept cases where b has unknown values but a is still deterministicly before or same as b', ->
-    DateTime.parse('2000-01-01T00:00:00.0').sameOrBefore(DateTime.parse('2000-01-01T00:00:00')).should.be.true()
-    DateTime.parse('2000-01-01T00:00:00.0').sameOrBefore(DateTime.parse('2000-01-01T00:00')).should.be.true()
-    DateTime.parse('2000-01-01T00:00:00.0').sameOrBefore(DateTime.parse('2000-01-01T00')).should.be.true()
-    DateTime.parse('2000-01-01T00:00:00.0').sameOrBefore(DateTime.parse('2000-01-01')).should.be.true()
-    DateTime.parse('2000-01-01T00:00:00.0').sameOrBefore(DateTime.parse('2000-01')).should.be.true()
-    DateTime.parse('2000-01-01T00:00:00.0').sameOrBefore(DateTime.parse('2000')).should.be.true()
+  it 'should return null for cases where b has unknown values but a is not deterministicly before or same as b', ->
```

Comment:
Again, description in the test cases isn't technically true.  In all the tests below, `a` _is_ deterministically before or same as `b` because `a` represents the earliest possible moment in the uncertainty range of `b` (e.g., there is no possible moment in `b` that could be `before` a`).